crates/verify/src/utils.rs

```diff
@@ -136,6 +137,56 @@ pub fn build_using_cache(
     eyre::bail!("couldn't find cached artifact for contract {}", args.contract.name)
 }
 
+pub fn get_immutable_refs(
+    artifact: &CompactContractBytecode,
+) -> Option<BTreeMap<String, Vec<Offsets>>> {
+    if artifact.deployed_bytecode.as_ref().is_some_and(|b| !b.immutable_references.is_empty()) {
+        let immutable_refs =
+            artifact.deployed_bytecode.as_ref().unwrap().immutable_references.clone();
+
+        return Some(immutable_refs);
+    }
+
+    None
+}
```

Comment:
nit: can we make this just return a `Vec` of `(start, end)` tuples so that we can just iterate over offsets in `extract_immutable_refs`